edit/edcore/insert.go

```diff
@@ -210,35 +212,47 @@ func (ed *editor) moveDotRight() {
 	ed.dot += w
 }
 
-func (ed *editor) moveDotLeftWord() {
+func (ed *editor) moveDotLeftIndexFunction(fn func(rune) bool) {
 	if ed.dot == 0 {
 		return
 	}
 	space := strings.LastIndexFunc(
-		strings.TrimRightFunc(ed.buffer[:ed.dot], unicode.IsSpace),
-		unicode.IsSpace) + 1
+		strings.TrimRightFunc(ed.buffer[:ed.dot], fn),
+		fn) + 1
 	ed.dot = space
 }
 
-func (ed *editor) moveDotRightWord() {
+func (ed *editor) moveDotRightIndexFunction(fn func(rune) bool) {
 	// Move to first space
-	p := strings.IndexFunc(ed.buffer[ed.dot:], unicode.IsSpace)
+	p := strings.IndexFunc(ed.buffer[ed.dot:], fn)
 	if p == -1 {
 		ed.dot = len(ed.buffer)
 		return
 	}
 	ed.dot += p
 	// Move to first nonspace
-	p = strings.IndexFunc(ed.buffer[ed.dot:], notSpace)
+	p = strings.IndexFunc(ed.buffer[ed.dot:], func(r rune) bool { return !fn(r) })
 	if p == -1 {
 		ed.dot = len(ed.buffer)
 		return
 	}
 	ed.dot += p
 }
 
-func notSpace(r rune) bool {
-	return !unicode.IsSpace(r)
+func (ed *editor) moveDotLeftWord() {
+	ed.moveDotLeftIndexFunction(unicode.IsSpace)
+}
+
+func (ed *editor) moveDotRightWord() {
+	ed.moveDotRightIndexFunction(unicode.IsSpace)
+}
+
+func (ed *editor) moveDotLeftSmallWord() {
+	ed.moveDotLeftIndexFunction(isAlnum)
```

Comment:
There is something curious here; the internal functions are given a function that returns whether a rune is a separator above, but here it is the opposite: they are given a function that returns whether a rune is **not** a separator.

Both would in fact work, but for clarify, please define a function `isNotAlnum` and use that instead.